tests/src/test/scala/whisk/core/container/test/ContainerUtilsTests.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.container.test
+
+import org.junit.runner.RunWith
+import org.scalatest.fixture.FlatSpec
+import org.scalatest.junit.JUnitRunner
+import java.io.File
+
+import common.StreamLogging
+
+import whisk.common.TransactionId
+import whisk.core.container.ContainerUtils
+import whisk.common.Logging
+import whisk.core.container.ContainerHash
+import org.scalatest.BeforeAndAfter
+import java.io.FileWriter
+import java.nio.charset.StandardCharsets
+import org.scalatest.Matchers
+import org.scalatest.concurrent.TimeLimitedTests
+import org.scalatest.time.SpanSugar._
+
+/**
+ * Unit tests for ContainerPool and, by association, Container and WhiskContainer.
+ */
+@RunWith(classOf[JUnitRunner])
+class ContainerUtilsTests extends FlatSpec
+    with BeforeAndAfter
+    with Matchers
+    with StreamLogging
+    with TimeLimitedTests {
+
+    val timeLimit = 100 millis
+
+    before {
+        stream.reset()
+    }
+
+    implicit val transid = TransactionId.testing
+
+    behavior of "getDockerLogContent"
+
+    case class FixtureParam(file: File, writer: FileWriter, cu: ContainerUtilsTester)
+
+    def withFixture(test: OneArgTest) = {
+        val file = File.createTempFile(this.getClass.getName, test.name.replaceAll("[^a-zA-Z0-9.-]", "_"))
+        val writer = new FileWriter(file)
+        val cu = new ContainerUtilsTester(file)
+
+        val fixture = FixtureParam(file, writer, cu)
+
+        try {
+            withFixture(test.toNoArgTest(fixture))
+        } finally {
+            writer.close()
+            file.delete()
+        }
+    }
```

Comment:
I was using fixtures because they provide a central place for setting and - particularly - cleaning up. The fixture makes sure that temporary files are closed and removed after each test.

All this is also possible with the approach you suggest above - but requires a `try ... finally`bracket around each single test.